src/hardware/ad7172.rs

```diff
@@ -0,0 +1,225 @@
+// (AD7172 https://www.analog.com/media/en/technical-documentation/data-sheets/AD7172-2.pdf)
+
+use core::fmt::Debug;
+
+use defmt::info;
+
+use super::hal::hal::{
+    blocking::{
+        delay::DelayUs,
+        spi::{Transfer, Write},
+    },
+    digital::v2::OutputPin,
+};
+
+// ADC Register Adresses
+#[allow(unused)]
+pub enum AdcReg {
+    STATUS = 0x00,
+    ADCMODE = 0x1,
+    IFMODE = 0x2,
+    DATA = 0x04,
+    ID = 0x7,
+    FILTCON0 = 0x28,
+    FILTCON1 = 0x29,
+    FILTCON2 = 0x2a,
+    FILTCON3 = 0x2b,
+    CH0 = 0x10,
+    CH1 = 0x11,
+    CH2 = 0x12,
+    CH3 = 0x13,
+    SETUPCON0 = 0x20,
+    SETUPCON1 = 0x21,
+    SETUPCON2 = 0x22,
+    SETUPCON3 = 0x23,
+    OFFSET0 = 0x30,
+    OFFSET1 = 0x31,
+    OFFSET2 = 0x32,
+    OFFSET3 = 0x33,
+    GAIN0 = 0x38,
+    GAIN1 = 0x39,
+    GAIN2 = 0x3a,
+    GAIN3 = 0x3b,
+}
```

Comment:
I wonder why we use the empty struct. It provides no protection through type checking or safety since the final type is `u32` always.

And then not duplicating the field name each of its values would be nice. This is just one more line of code:

```rust

pub mod Adcmode {
    pub mod Ref {
        pub const ENABLED: u32 = 1 << 15; // Internal reference enable
    }
    pub mod Mode {
        pub const CONTINOUS_CONVERSION: u32 = 0b000 << 4; // Continuous conversion mode
    }
    pub mod Clocksel {
        pub const INTERNAL_OSC: u32 = 0b00 << 2; // Internal oscillator clock source
        pub const INTERNAL_OSC_OUT: u32 = 0b01 << 2; // Internal oscillator clock source and output
        pub const EXTERNAL_CLOCK: u32 = 0b10 << 2; // External clock input
        pub const EXTERNAL_OSC: u32 = 0b11 << 2; // External oscillator clock source
    }
}
```

What do you think?